File: solutions/3451-string-compression-iii/solution.java
class Solution {
    public String compressedString(String word) {
        StringBuilder comp = new StringBuilder();
        int left = 0;
        while (left < word.length()) {
            int count = 0;
            char curr = word.charAt(left);

            while (left < word.length() && count < 9 && word.charAt(left) == curr) {
                count++;
                left++;
            }
            comp.append(count).append(curr);
        }

        return comp.toString();
    }
}

File: solutions/3451-string-compression-iii/solution.js
/**
 * @param {string} word
 * @return {string}
 */
var compressedString = function(word) {
    let res = '', count = 1;
    for (let i = 1; i <= word.length; i++) {
        if (word[i] === word[i - 1] && count < 9) count++;
        else res += count + word[i - 1], count = 1;
    }
    return res;
};
